server/sonar-process/src/main/java/org/sonar/process/LogbackHelper.java

```diff
@@ -254,18 +259,50 @@ public boolean isAllLogsToConsoleEnabled(Props props) {
    *
    * @throws IllegalArgumentException if the value of {@link #SONAR_LOG_LEVEL_PROPERTY} is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
    */
-  public Level configureRootLogLevel(Props props) {
-    return configureRootLogLevel(props, SONAR_LOG_LEVEL_PROPERTY);
+  public Level configureRootLogLevel(Props props, ProcessId processId) {
+    return configureRootLogLevel(props,
+      SONAR_LOG_LEVEL_PROPERTY,
+      SONAR_PROCESS_LOG_LEVEL_PROPERTY.replace(PROCESS_NAME_PLACEHOLDER, processId.getKey()));
   }
 
   /**
-   * Configure the log level of the root logger reading the value of specified property.
+   * Configure the log level of the root logger reading the value of specified properties. The lowest level will be
+   * applied.
    *
    * @throws IllegalArgumentException if the value of the specified property is not one of {@link #ALLOWED_ROOT_LOG_LEVELS}
    */
-  public Level configureRootLogLevel(Props props, String propertyKey) {
-    Level newLevel = Level.toLevel(props.value(propertyKey, Level.INFO.toString()), Level.INFO);
-    return configureRootLogLevel(newLevel);
+  public Level configureRootLogLevel(Props props, String... propertyKeys) {
+    return configureLoggerLogLevel(getRootContext().getLogger(ROOT_LOGGER_NAME), props, propertyKeys);
+  }
+
+  private Level configureLoggerLogLevel(Logger logger, Props props, String... propertyKeys) {
+    Level newLevel = Level.INFO;
+    for (String propertyKey : propertyKeys) {
+      Level level = getPropertyValueAsLevel(props, propertyKey);
+      if (!level.isGreaterOrEqual(newLevel)) {
+        newLevel = level;
+      }
+    }
+    logger.setLevel(newLevel);
+    return newLevel;
+  }
+
+  private static Level getPropertyValueAsLevel(Props props, String propertyKey) {
+    Level level = Level.toLevel(props.value(propertyKey, Level.INFO.toString()), Level.INFO);
+    if (!isAllowed(level)) {
+      throw new IllegalArgumentException(String.format("log level %s in property %s is not a supported value (allowed levels are %s)",
+        level, propertyKey, Arrays.toString(ALLOWED_ROOT_LOG_LEVELS)));
+    }
+    return level;
+  }
+
+  private static boolean isAllowed(Level level) {
+    for (Level allowedRootLogLevel : ALLOWED_ROOT_LOG_LEVELS) {
+      if (level == allowedRootLogLevel) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)